src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -72,14 +77,30 @@ class CollectorService(
 
         val limitSize = configuration.messageCacheSize
         mqSubject = PublishSubject.create()
-
-        subscriberMonitor = subscribe(MessageListener { _: DeliveryMetadata, batch: MessageBatch -> mqSubject.onNext(batch) })
-        streamObservable = mqSubject.flatMapIterable(MessageBatch::getMessagesList)
-                .groupBy { message ->
-                    message.metadata.id.run {
-                        SessionKey(message.metadata.id.bookName, connectionId.sessionAlias, direction)
-                    }.also(BufferMetric::processMessage)
+        protoSubscriberMonitor =
+            checkNotNull(protoMessageRouter.subscribeAll({ _: DeliveryMetadata, batch: MessageBatch ->
+                batch.messagesList.forEach {
+                    mqSubject.onNext(ProtoMessageWrapper(it))
                 }
+            })) { "Can not subscribe for listening protobuf messages" }
+        transportSubscriberMonitor =
+            checkNotNull(transportMessageRouter.subscribeAll({ _: DeliveryMetadata, batch: GroupBatch ->
```

Comment:
The current implementation of th2 transport parsed message doesn't support null value `body: Map<String, Any>`. Do we want to implement this feature?

Reply:
I think we MUST do that

Reply:
Implemented both request:

1. `This might cause an exception if no pins with transport-group are specified. So, maybe, we should make the subscription optional for transport protocol`
2. `The current implementation of th2 transport parsed message doesn't support null value body: Map<String, Any>. Do we want to implement this feature?` ... `I think we MUST do that`